sisyphus/localengine.py

```diff
@@ -86,6 +86,9 @@ def __init__(self, cpus=1, gpus=0, **kwargs):
         self.max_resources = {'cpu': cpus, 'gpu': gpus}
         self.max_resources.update(kwargs)
         self.free_resources = self.max_resources.copy()
+        cuda_visible_devices = os.environ.get("CUDA_VISIBLE_DEVICES", "")
+        assert gpus == 0 or len(cuda_visible_devices.split(",")) == gpus
```

Comment:
why the `or` statement?

Reply:
For zero gpus  `cuda_visible_devices` will be empty string, so `cuda_visible_devices.split(",")` will be `[""]` of length 1